core/src/main/scala/akka/kafka/internal/CommittingProducerSinkStage.scala

```diff
@@ -164,12 +167,9 @@ private final class CommittingProducerSinkStageLogic[K, V, IN <: Envelope[K, V,
     case CommittingProducerSinkStage.CommitNow => commit(Interval)
   }
 
-  private def collectOffset(count: Int, offset: Committable): Unit = {
-    awaitingProduceResult -= count
-    offsetBatch = offsetBatch.updated(offset)
-    if (offsetBatch.batchSize >= stage.committerSettings.maxBatch) commit(BatchSize)
+  private def collectOffset(offset: Committable): Unit =
+    if (updateBatch(offset)) commit(BatchSize)
```

Comment:
I believe the `awaitingProduceResult -= count` could stay here.

Reply:
I moved the update of this counter to each offset callback.